hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java

```diff
@@ -341,7 +341,7 @@ private void handleOperationError(Operation operation, Throwable e) {
 
         operation.logError(e);
 
-        if (e instanceof ResponseNotSentException || !operation.returnsResponse()) {
+        if (e instanceof ResponseNotSentException || !operation.isRetryable()) {
```

Comment:
Imagine an operation which is not retryable but returnsResponse=true, it will not be able to return response.
I see that currently `isRetryable`=`returnsResponse` except `PartitionCheckIfLoadedOperation` but I think we should address that case too.